node/src/blockchain/blockchain_interface/blockchain_interface_web3/lower_level_interface_web3.rs

```diff
@@ -1,395 +1,762 @@
-        CONTRACT_ABI, REQUESTS_IN_PARALLEL,
-    };
-    use crate::blockchain::blockchain_interface::lower_level_interface::{LowBlockchainInt, ResultForBalance};
-    use crate::blockchain::blockchain_interface::BlockchainError;
+    use crate::blockchain::blockchain_interface::{BlockchainError, BlockchainInterface};
     use crate::sub_lib::wallet::Wallet;
-    use crate::test_utils::http_test_server::TestServer;
-    use crate::test_utils::make_paying_wallet;
-    use ethereum_types::U64;
-    use masq_lib::blockchains::chains::Chain;
-    use masq_lib::test_utils::utils::TEST_DEFAULT_CHAIN;
     use masq_lib::utils::find_free_port;
-    use serde_json::{json, Value};
-    use std::net::Ipv4Addr;
-    use std::rc::Rc;
     use std::str::FromStr;
-    use std::sync::{Arc, Mutex};
-    use web3::contract::Contract;
-    use web3::transports::{Batch, Http};
-    use web3::types::U256;
-    use web3::{BatchTransport, Web3};
-    use crate::blockchain::test_utils::TestTransport;
+    use ethereum_types::{H256, U64};
+    use futures::Future;
+    use web3::types::{BlockNumber, Bytes, FilterBuilder, H2048, Log, TransactionReceipt, U256};
+    use masq_lib::test_utils::mock_blockchain_client_server::MBCSBuilder;
+    use crate::blockchain::blockchain_interface::blockchain_interface_web3::TRANSACTION_LITERAL;
+    use crate::blockchain::blockchain_interface::blockchain_interface_web3::lower_level_interface_web3::TransactionReceiptResult;
+    use crate::blockchain::blockchain_interface::data_structures::errors::BlockchainError::QueryFailed;
+    use crate::blockchain::test_utils::{make_blockchain_interface_web3, make_tx_hash, ReceiptResponseBuilder};
+    use crate::test_utils::{assert_string_contains, make_wallet};
 
     #[test]
-    fn low_interface_web3_transaction_fee_balance_works() {
+    fn get_transaction_fee_balance_works() {
         let port = find_free_port();
-        let test_server = TestServer::start(
-            port,
-            vec![br#"{"jsonrpc":"2.0","id":0,"result":"0xDEADBEEF"}"#.to_vec()],
-        );
-        let (_event_loop_handle, transport) = Http::with_max_parallel(
-            &format!("http://{}:{}", &Ipv4Addr::LOCALHOST.to_string(), port),
-            REQUESTS_IN_PARALLEL,
-        )
-        .unwrap();
-        let chain = TEST_DEFAULT_CHAIN;
-        let subject = make_subject(transport, chain);
+        let _blockchain_client_server = MBCSBuilder::new(port)
+            .response("0x23".to_string(), 1)
+            .start();
+        let wallet = &Wallet::from_str("0x3f69f9efd4f2592fd70be8c32ecd9dce71c472fc").unwrap();
+        let subject = make_blockchain_interface_web3(Some(port));
+
+        let result = subject
+            .lower_interface()
+            .get_transaction_fee_balance(wallet.address())
+            .wait();
+
+        assert_eq!(result, Ok(35.into()));
+    }
+
+    #[test]
+    fn get_transaction_fee_balance_returns_an_error_for_unintelligible_response_to_requesting_eth_balance(
```

Comment:
you can shorten the name to `get_transaction_fee_balance_returns_an_error_for_unintelligible_response` and it will preserve the same meaning